pkg/proc/threads.go

```diff
@@ -325,6 +340,51 @@ func next(dbp Process, stepInto, inlinedStepOut bool) error {
 	return nil
 }
 
+func setStepIntoBreakpoints(dbp Process, text []AsmInstruction, topframe Stackframe, sameGCond ast.Expr) error {
+	for _, instr := range text {
+		if instr.Loc.File != topframe.Current.File || instr.Loc.Line != topframe.Current.Line || !instr.IsCall() {
+			continue
+		}
+
+		if instr.DestLoc != nil && instr.DestLoc.Fn != nil {
+			if err := setStepIntoBreakpoint(dbp, []AsmInstruction{instr}, sameGCond); err != nil {
+				return err
+			}
+		} else {
+			// Non-absolute call instruction, set a StepBreakpoint here
+			if _, err := allowDuplicateBreakpoint(dbp.SetBreakpoint(instr.Loc.PC, StepBreakpoint, sameGCond)); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+var errStepShouldStepInstruction = errors.New("step should step by one instruction")
+
+func setStepIntoBreakpointsReverse(dbp Process, text []AsmInstruction, topframe Stackframe, sameGCond ast.Expr) error {
+	// Set a breakpoint after every CALL instruction
+	for i, instr := range text {
+		if instr.Loc.File != topframe.Current.File || !instr.IsCall() || instr.DestLoc == nil || instr.DestLoc.Fn == nil {
+			continue
+		}
+
+		if fn := instr.DestLoc.Fn; strings.HasPrefix(fn.Name, "runtime.") && !isExportedRuntime(fn.Name) {
+			continue
+		}
+
+		if i+1 < len(text) {
+			if topframe.Current.PC == text[i+1].Loc.PC {
+				return errStepShouldStepInstruction
```

Comment:
I'm not crazy about communicating internally via errors like this.

Reply:
Changed it to a check inside `Step`.